Guest accounts feature

jespino opened the pull request:
#### Summary
Guest accounts feature

mattermod commented:
`Setup Test Server` label detected. Spinmint test server created if build succeeds (checks pass and no conflicts with base branch).

mattermod commented:
Spintmint test server creation failed. See the logs for more information.

mattermod commented:
`Setup Test Server` label detected. Spinmint test server created if build succeeds (checks pass and no conflicts with base branch).

mattermod commented:
Spinmint test server created at: https://i-0135dc6be4b3c626f.test.spinmint.com

Test Admin Account: Username: `sysadmin` | Password: `sysadmin`

Test User Account: Username: `user-1` | Password: `user-1`

Internal IP: 172.26.24.16
Instance ID: i-0135dc6be4b3c626f

mattermod commented:
Spinmint test running for more than 7 days. This test server was terminated.

mattermod commented:
`Setup Test Server` label detected. Spinmint test server created if build succeeds (checks pass and no conflicts with base branch).

mattermod commented:
Spinmint test server created at: https://i-00ad8211ccdeed1d4.test.spinmint.com

Test Admin Account: Username: `sysadmin` | Password: `sysadmin`

Test User Account: Username: `user-1` | Password: `user-1`

Internal IP: 172.26.38.216
Instance ID: i-00ad8211ccdeed1d4

ogi-m commented:
### Invite flow issues: 

**1. "Copy Link" button doesn't work**
- Go to Invite People > Invite Members 
- Click "Copy Link" button next to the invite link
Observed: Text isn't copied and there's no visual confirmation that the button was clicked
Expected: Text is copied and there's visual confirmation that the button was clicked

**2. Blank screen when inviting new Members/Guests** 
1. Go to Invite People > Invite Members (same issue happens on Invite Guests)
2. Enter an email that doesn't exist in the system and click "Invite Members"
Observed: Screen goes blank (see screenshot for console errors)
Expected: Confirmation screen is shown saying an invite was sent to the email address

![Screenshot (167)](https://user-images.githubusercontent.com/15314183/61080188-c73dba00-a424-11e9-8a9c-df0fa1b20c11.png)

**3. Invite modal can't be scrolled**
- Go to Invite People > Invite Guests
- Click "Set a custom message" to expand the "Custom message" field
Observed: "Invite Guests" button is pushed out out of view with no way to scroll it into view
Expected: User can scroll the modal

**Note**: 1080p display at 125% UI scaling


**4. Existing Guests are not added to the team**
- Create a team and go to Invite People > Invite Guests
- Select '@guest' user who is not a guest on the current team
- Select a channel for the guest user to join and click "Invite Guests"
- Success message is shown saying that the guest was added to the team
- Close the modal and view the members of the channel (and/or the team) the guest was added to
Observed: Guest user was not added to the channel or the team
Expected: The user is added to the team and to the specified channel

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Spintmint test server creation failed. See the logs for more information.

mattermod commented:
New commit detected. SpinWick upgrade will occur after the build is successful.

mattermod commented:
Spinmint test server destroyed

mattermod commented:
Error doing the upgrade process. Please remove the label and try again.

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Spintmint test server creation failed. See the logs for more information.

mattermod commented:
Mattermost test server created! :tada:

Access here: https://mattermost-server-pr-11428.test.mattermost.cloud

Test Admin Account: Username: `sysadmin` | Password: `Sys@dmin123`

Test User Account: Username: `user-1` | Password: `User-1@123`

mattermod commented:
Spinmint test server destroyed

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Mattermost test server created! :tada:

Access here: https://mattermost-server-pr-11428.test.mattermost.cloud

Test Admin Account: Username: `sysadmin` | Password: `Sys@dmin123`

Test User Account: Username: `user-1` | Password: `User-1@123`

jespino commented:
@ogi-m 
1. I can reproduce it locally
2. I fixed that here: https://github.com/mattermost/mattermost-webapp/pull/3102
3. This is that way by design I think <- @michaelgamble what do you think about the point 3?
4. Clearly something is wrong there, I have to check it properly (in some cases work, but not in all cases).
5. I'll check it
6. I'll check it
7. The current mattermost email validation is just that, we can be more strict about it, but have other implications.
8. @michaelgamble which is the good one here?
9. I checked that with @michaelgamble and I think the current styling is the correct one. Can you confirm that @michaelgamble?
10. I'll fix that
11. Yes, that is a know issue, I have to fix that too.
12. I don't agree with that, when you are inviting guest if you put regular users, the most important failure is that you can't add regular users as guests. @michaelgamble what do you think about it?
13. I'll fix that

ogi-m commented:
@jespino I've updated issues 8, 9 and 12 with links to the design for reference.
> 1. I can't reproduce it locally

This one is a blocker - I can't create any guest users (And I can't reset the password for @guest).

michaelgamble commented:
@jespino and @ogi-m 
3. I would expect the whole page to scroll with the exception of the "back" and "close" buttons would stay fixed on the page while scrolling. (scroll bar would be placed full height of screen right side)
8. Use the titles in the invision design as those were the ones that went through final reviews. "Invite People"
9. @ogi-m can you reach out to me on this one, I'm wondering if its the difference of which platform you are viewing in, but because I'm not sure i wouldn't mind doing a quick zoom to take a look.
12. The intention here is, the invite as a guest failed, however they most likely follow up action is "oh shoot I want to add them to the channel anyways" so thats why out of a few rounds of review and discussion we added that use case to the design. Lets stick with whats in invision on this one for now.

mattermod commented:
New commit detected. SpinWick upgrade will occur after the build is successful.

mattermod commented:
New commit detected. SpinWick upgrade will occur after the build is successful.

mattermod commented:
Mattermost test server updated!

Access here: https://mattermost-server-pr-11428.test.mattermost.cloud

mattermod commented:
`Setup Test Server` label detected. Spinmint test server created if build succeeds (checks pass and no conflicts with base branch).

mattermod commented:
Spintmint test server creation failed. See the logs for more information.

mattermod commented:
Spintmint test server creation failed. See the logs for more information.

michaelgamble commented:
@jespino 
9. reviewed with @ogi-m seems the weights are off here, it may have to do with the fact that on the previous screen its a mix of font-weight 300 and 400.. but on this screen it should be 'normal' which calculates to 400

Also I noticed on @ogi-m screenshots another issue of the input field heights seem to be wrong in chrome in windows - a discrepancy, we'll have to do a UX review pass in windows and see if anything else is out of place too.

ogi-m commented:
Two more issues with the invite flow:

14. "Details'  text overlaps with a longer email
- Go to Invite People > Invite People
- Enter a longer email and click "Invite Members"
- Note the email and the "Details" text in the table on the summary page
Observed: Email and text overlap (see attached screenshot below)
Expected: No overlap

15. Wrong green color in dark mode
- Go to Account Settings > Display > Theme and select Mattermost Dark
- Go to Invite People > Invite People
- Enter an email and click "Invite Members"
- Note the green color in the table header
Observed: Different green from [the one in invision design](https://projects.invisionapp.com/d/main/default/?origin=v7#/console/17561763/364046127/preview)
Expected: Green matches invision
![Screenshot (172)](https://user-images.githubusercontent.com/15314183/61248004-7d214500-a752-11e9-9414-5f18203ba55a.png)

mattermod commented:
New commit detected. SpinWick upgrade will occur after the build is successful.

mattermod commented:
Mattermost test server updated!

Access here: https://mattermost-server-pr-11428.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick upgrade will occur after the build is successful.

mattermod commented:
Mattermost test server updated!

Access here: https://mattermost-server-pr-11428.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick upgrade will occur after the build is successful.

mattermod commented:
Spintmint test server creation failed. See the logs for more information.

srkgupta commented:
Hi @jespino 

Observed an issue while testing the Guest Accounts feature. Can you please check and fix the same. 

**Whitelisted Domains for Guest Accounts does not works**
Steps to reproduce this issue:
- Navigate to System Console -> Guest Access Page. 
- Enter the Whitelisted Guest domains as "company.com"
- Now trigger the Guest Invitation process from the main screen and try to signup with an email belonging to a different domain other than the one provided in the whitelisted section
- User is allowed to create an account and login as Guest without any restriction.

srkgupta commented:
Hi @jespino, 

Some other issues I observed during my initial testing:

1) When Guest account is turned off from the System Console, the option to Invite users as Guest is still shown in the main screen. Only when user clicks on it and proceeds to the next step, a message is displayed indicating that the Guest account featured needs to be turned on. 
2) Similarly if license is not uploaded on an instance, the option to invite users as Guest is still shown in the main screen. Only when user clicks on it and proceeds to the next step, a message is displayed indicating that the License needs to be uploaded in order to use the Guest Accounts feature. This might be problematic to GA instances without Enterprise licenses. We generally turn off the features which are not available, rather than showing an error message.

wiersgallak commented:
@jespino Can we please also make sure this feature is labelled as "(Beta)" on the top of the Guest Access page and the left hand side listing in the Console?

srkgupta commented:
Hi @jespino, 

Few other issues observed:

**1) Unable to Enforce MFA only for Guest Users**
	- Login to System Console and ensure MFA is enabled and Enforce MFA is set to false. 
	- Navigate to Guest Access page and check if Admin is allowed to Enforce MFA for Guest Users. 
	- This option is disabled and thus we are unable to enforce MFA only for Guest Users. 
	- Expectations: Since MFA is enabled, the option to enforce MFA for Guest Users should also be enabled. 
		
**2) Unable to complete MFA Process. Getting an Empty Response and an error "Failed to fetch"**
	- Login to System Console and ensure MFA is enabled. 
	- Set Enforce MFA to true and save.
	- User will be redirected to Configure MFA page. Try to scan the QR Code and submit the token. 
	- The PUT request sent to `/api/v4/users/<USER_ID>/mfa` endpoint receives an EMPTY_RESPONSE and an error `Failed to fetch` is displayed in the UI.

jespino commented:
14. @michaelgamble what do you thing should be the behavior there? ellipsis? Is the same behavior for users?

jespino commented:
15. Fixed

jespino commented:
The 1 from @srkgupta about MFA I think is the expected behavior, can you confirm that @wiersgallak?

jespino commented:
About @ogi-m bugs:

1. Fixed
2. Fixed
3. Fixed
4. Still need to check that
5. Fixed
6. Fixed
7. Wont fix
8. Fixed
9. Fixed
10. Fixed
11. Fixed
12. Fixed
13. Pending
14. Pending
15. Fixed

srkgupta commented:
Hi @jespino 

Regarding the issue `Unable to Enforce MFA only for Guest Users`, if you look into this [design](https://projects.invisionapp.com/share/7POW2GOGFVT#/screens/369029392), a message is displayed under the field 'Enforce Multi-factor authentication' indicating the reason why it's disabled, i.e. `MFA is currently not enabled`. Hence it is evident that if MFA is enabled, the option to Enforce MFA for Guest Users should be enabled as well. 

Anyways will wait for a confirmation from @wiersgallak

jespino commented:
@srkgupta Domain restrictions fixed here: https://github.com/mattermost/mattermost-server/pull/11660

jespino commented:
@srkgupta From your first numered list:

1 and 2 fixed here: https://github.com/mattermost/mattermost-webapp/pull/3158
3. Was a decision made after the design, support demote to easy the initial steps of adoption of the feature.
4 and 5 are part of another feature that the cloud team is working on, after that feature is implemented this will be implemented. So "wont fix" for now.

wiersgallak commented:
> The 1 from @srkgupta about MFA I think is the expected behavior, can you confirm that @wiersgallak?

@srkgupta & @jespino Yes, our spec was for when MFA enable is true and MFA enforce is true for regular users, then the option to Enforce MFA for guests would be available.  MFA should be optional  when enable MFA is true and MFA enforce is false.  We will need to use the updated message when MFA enabled = false under Authentication > MFA: 

>Regarding the issue Unable to Enforce MFA only for Guest Users, if you look into this design, a message is displayed under the field 'Enforce Multi-factor authentication' indicating the reason why it's disabled, i.e. MFA is currently not enabled. Hence it is evident that if MFA is enabled, the option to Enforce MFA for Guest Users should be enabled as well.

srkgupta commented:
Thanks for the update @jespino and thanks for the confirmation @wiersgallak

srkgupta commented:
Hi @jespino 

Few more issues observed:

**11) Guest user is not able to initiate a Direct Message chat with anyone in the Channel**
- Login as a Guest User and click on + icon next to Direct Messages
- Select a user to initiate a Direct Message chat
- Guest user is not allowed to initiate a chat and an error is thrown in the JS console as shown in this screenshot

![DirectMessagesChat](https://user-images.githubusercontent.com/1429138/61478360-fcab5000-a9ae-11e9-8483-671af52361fc.PNG)

**12) Searching through Channels shows infinite loading and a JS error**
- Login as a Guest user and use the shortcut key to Switch Channels
- Type in something like Guest and check the console 
- As shown in this screenshot, an error is shown and the UI shows an infinite loading. 

![SearchChannel2](https://user-images.githubusercontent.com/1429138/61478369-02089a80-a9af-11e9-850b-383c5c8716be.PNG)
![SearchChannel](https://user-images.githubusercontent.com/1429138/61478373-046af480-a9af-11e9-9929-6e6c4eecaed8.PNG)

**13) Updating the Channel Header does not reflects for the Guest User**
- As shown in these screenshots, a SysAdmin had updated the Channel Header
- However, the Guest user was not able to see the Channel Header (as shown in this screenshot)
- Only after sometime (not sure when), the channel header started appearing
![Guest_ChannelHeader](https://user-images.githubusercontent.com/1429138/61478377-06cd4e80-a9af-11e9-8407-0690e588e6df.PNG)

mattermod commented:
Spinmint test running for more than 7 days. This test server was terminated.

srkgupta commented:
Hi @jespino, 

The build is broken and we are completely blocked because of it. Can you please check and fix this issue ASAP. 

https://build.mattermost.com/blue/organizations/jenkins/mp%2Fmattermost-server-pr-new/detail/PR-11428/29/pipeline

jespino commented:
@srkgupta Sure, I'll check it now, it only needs just a merge from master and fixing certain things

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

cpanato commented:
I will look why failed

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Spintmint test server creation failed. See the logs for more information.

cpanato commented:
im debugging

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Spintmint test server creation failed. See the logs for more information.

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Mattermost test server created! :tada:

Access here: https://mattermost-server-pr-11428.test.mattermost.cloud

| Account Type | Username | Password |
|---|---|---|
| Admin | sysadmin | Sys@dmin123 |
| User | user-1 | User-1@123 |

cpanato commented:
@jespino @srkgupta fixed

jespino commented:
@cpanato Thanks!

mattermod commented:
New commit detected. SpinWick upgrade will occur after the build is successful.

mattermod commented:
Mattermost test server updated!

Access here: https://mattermost-server-pr-11428.test.mattermost.cloud

ogi-m commented:
**Guests labelled as "Members" on User Configuration**
- Log in as an admin and go to System Console > User Management
- Click on a guest user
- Note the "Role"
Observed: Role is listed as "Member"
Expected: Role should be "Guest"

![2019-07-19_1701](https://user-images.githubusercontent.com/15314183/61545955-df2ac480-aa48-11e9-9ee8-bfa9a60c9fef.png)

jespino commented:
Updates:

@ogi-m:

1. Fixed
2. Fixed
3. Fixed
4. Fixed
5. Fixed
6. Fixed
7. Wont fix
8. Fixed
9. Fixed
10. Fixed
11. Fixed
12. Fixed
13. Fixed
14. Waiting for @michaelgamble response
15. Fixed
16. Pending

@srkgupta:

1. Fixed
2. Fixed
3. Working as expected
4. Wont fix for now
5. Wont fix for now
6. Wont fix for now
7. Working as expected
8. I think is not related to the PR
9. Decided to defer it until phase2
10. Fixed
11. Can not reproduce
12. Pending
13. Fixed
14. Fixed
15. Can not reproduce

michaelgamble commented:
@jespino @ogi-m 
14. Hmm this is a tough one.. I think allow the left column to expand past its 50% width shrinking the right column proportionally.. causing the contents of the right column to line wrap as much as needed.. I think its really important to preserve seeing the email in full in this summary screen... also keep in mind the column headers need to be adjusting in width along with it.

mattermod commented:
Spinmint test server destroyed

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Spintmint test server creation failed. See the logs for more information.

cpanato commented:
@jespino @srkgupta you dont need to remove and re-add the label to get a new test server, every new push will update the test server, this avoids errors as well

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

srkgupta commented:
Hi @cpanato 

Sure. We had issues with the previous build. The latest changes were not picked up from the mattermost-webapp repository and hence wanted to give this a try, i.e. to remove the label, destroy the instance and initiate the instance again.

cpanato commented:
this is another issue, which i will look later

srkgupta commented:
Ok Thanks for letting me know. Will take care going forward.

cpanato commented:
it get the branch but maybe an old one
```
+ git checkout PR-11428

error: pathspec 'PR-11428' did not match any file(s) known to git.

+ git checkout feature-guest-accounts

Previous HEAD position was f0467fa... Fix failing E2E test due to recent DOM structure change (#3182)

Switched to a new branch 'feature-guest-accounts'

Branch feature-guest-accounts set up to track remote branch feature-guest-accounts from origin.

+ rm -rf ./dist

+ git rev-parse HEAD

+ export WEBAPP_GIT_COMMIT=ecd76edf876324fdb157422a06e1fdad90eeb4bf

+ echo Webapp Commit: ecd76edf876324fdb157422a06e1fdad90eeb4bf

Webapp Commit: ecd76edf876324fdb157422a06e1fdad90eeb4bf

+ curl -f -o ./dist.tar.gz https://releases.mattermost.com/mattermost-webapp/commit/ecd76edf876324fdb157422a06e1fdad90eeb4bf/mattermost-webapp.tar.gz

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current

                                 Dload  Upload   Total   Spent    Left  Speed


  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  8 26.0M    8 2175k    0     0  10.3M      0  0:00:02 --:--:--  0:00:02 10.3M
100 26.0M  100 26.0M    0     0  31.5M      0 --:--:-- --:--:-- --:--:-- 31.5M

+ mkdir ./dist

+ tar -xvf ./dist.tar.gz -C ./dist --strip-components=1

client/16.c7a3424b9fba68aede1f.js

client/0.ffa8cb46a7f5b15518c5.js

client/12.bc85a8711b21da40f9e7.js

client/20.4f89ed4770e9ad7d98fd.js.map

client/icon_60x60.png

client/8.76f2391c2b3e9ae6c673.js.map

client/15.6911e12e5ccf3fb05be4.js.map

client/33.26dc9c4da4780cbb70a7.js.map

client/icon_57x57.png

client/21.44752c8d8840ee94d162.js.map

client/images/

client/images/warning.png

client/images/img_trans.gif

client/images/appIcons.png

client/images/logo-email.png

client/images/favicon/
....
```

mattermod commented:
Mattermost test server created! :tada:

Access here: https://mattermost-server-pr-11428.test.mattermost.cloud

| Account Type | Username | Password |
|---|---|---|
| Admin | sysadmin | Sys@dmin123 |
| User | user-1 | User-1@123 |

cpanato commented:
@jespino @srkgupta if you want to the server get the latest webapp please rebase this branch or ammend a commit or whatever, so it will build again

srkgupta commented:
@jespino looks like the latest changes are still not deployed correctly. Can you please check. I am still able to see some of the issues which were fixed, like Guest Access (Beta) is not shown in the System console, etc.

cpanato commented:
need to do what i mention above, when the job ran was another sha in the webapp part, to get the most up-to-date need to push a new commit here to trigger the build again and then the server update

mattermod commented:
New commit detected. SpinWick upgrade will occur after the build is successful.

cpanato commented:
now is building with a62247336fefa0f6d1598895063a036fd3e6500d from webapp which is the latest

mattermod commented:
Mattermost test server updated!

Access here: https://mattermost-server-pr-11428.test.mattermost.cloud

srkgupta commented:
Just an update regarding the fixed issues:

Issues 1, 2 ,10,13 and 14 reported by me were fixed and are working fine now. 

New issues observed:
16) Guest User is allowed to delete a post in a direct message even when this option is restricted in the Permission Schemes
- Login as a system admin and navigate to Permission Schemes and ensure Guest User does not has option to Delete Posts
- Login as a Guest User and initiate a direct message chat with one person. 
- Send a message and check if that message post has option to Delete. Try to delete and the message gets deleted
Expectations: Delete option should not be shown for posts by Guest users since this option is restricted as per Permission schemes. 

17) Guest users are still forced to setup MFA when Enforce MFA is turned on for Users and Enforce MFA is disabled for Guest Users
- Login as System Admin and navigate to MFA. Ensure MFA is turned on and is enforced for all users. 
- Navigate to Guest Access page and ensure MFA is not enforced 
- Invite a Guest User and login as Guest User
- Guest User is forced to Setup MFA immediately. 
Expectations: Since the option to enforce MFA for Guest Users is disabled, Guest Users should not be enforced to setup MFA.

ogi-m commented:
@jespino Regarding my issue #12 - message text is updated but there's still no option to add existing members to team/channels

jespino commented:
@ogi-m the 12 is deferred to the next phase

ogi-m commented:
**16. Custom message missing styling**
- As a system admin invite a guest user to join a team and include a custom message
- Go to inbox and open the invite email

**Observed**: Custom message in the email is shown immediately below the "Join Team" button and is missing the styling as seen in the [design](https://projects.invisionapp.com/d/main/default/?origin=v7#/console/17561763/364046309/preview)
**Expected**: Custom message styling matches the design

![2019-07-20_1355](https://user-images.githubusercontent.com/15314183/61578371-15724d80-aaf6-11e9-9712-6ffee235f31f.png)

**17. Guest is taken to Town Square on first log-in**
- As a system admin invite a guest user to join a team and don't add the guest to "Town Square"
- Go to inbox and open the invite link
- Complete the signup process and log in

**Observed**: Guest is taken to Town Square which lands them on "Channel Not Found" page if Town Square isn't included in the list of channels they were added to
**Expected**: Guest is taken to the first channel based on alphabetical order on first log-in
![2019-07-20_1347](https://user-images.githubusercontent.com/15314183/61578316-266e8f00-aaf5-11e9-86b4-c316827bc32b.png)

**18. Wrong message and not logged-out when removed from channels**
- Log in as a guest
- In a different browser log in as a system admin and remove the guest from all channels or from the team
- Go back to the browser where the guest user is logged-in

**Observed**: _"You don't belongs to any team and as a guest you can't join any team by yourself, you need to be invited or added by someone else."_
[Expected](https://projects.invisionapp.com/d/main/default/?origin=v7#/console/17561763/364046315/preview): _"Your guest account has no channels assigned. Please contact an administrator."_

![2019-07-20_1400](https://user-images.githubusercontent.com/15314183/61578429-fe802b00-aaf6-11e9-9258-23ecd32732df.png)

**19. System message doesn't specify that a user was added as a guest**
- As a system admin add a guest user to a channel
- Observe the system message in the channel

**Observed**: 'guest-username' added to the channel by 'admin-username'
**Expected** ([per design](https://projects.invisionapp.com/d/main/default/?origin=v7#/console/17561763/364046318/preview)): 'guest-username' added to the channel as a guest by 'admin-username'


**20. "Channel not found" page has "Back to Town Square" link**
- As guest user try to open a channel you previously had access to ( directly via URL for example)

**Observed**: Page shows " Back to Town Square" link.
In case the guest isn't added to Town Square, this link will actually take the guest to a channel that they do have access to. 
**Expected**: The wording for this link should probably be different since Town Square isn't a channel that Guests necessarily belong to? @michaelgamble @jespino

grundleborg commented on store/sqlstore/tokens_store.go:
```diff
@@ -23,7 +23,7 @@ func NewSqlTokenStore(sqlStore SqlStore) store.TokenStore {
 		table.ColMap("Token").SetMaxSize(64)
 		table.ColMap("Type").SetMaxSize(64)
-		table.ColMap("Extra").SetMaxSize(128)
+		table.ColMap("Extra").SetMaxSize(2048)
```
What happens if this table already exists?

grundleborg replied:
As per discussion on MM, this will be fixed with a migration post-merge.

mattermod commented:
New commit detected. SpinWick upgrade will occur after the build is successful.

mattermod commented:
New commit detected. SpinWick upgrade will occur after the build is successful.

mattermod commented:
Spintmint test server creation failed. See the logs for more information.

mattermod commented:
Spintmint test server creation failed. See the logs for more information.

mattermod commented:
New commit detected. SpinWick upgrade will occur after the build is successful.

mattermod commented:
Spinmint test server destroyed

mattermod commented:
@jespino
Error trying doing the automated Cherry picking. Please do this manually

```
+++ Updating remotes...
Fetching upstream
Fetching origin
+++ Updating remotes done...
+++ Creating local branch automated-cherry-pick-of-#11428-upstream-release-5.14-1563826422
Branch 'automated-cherry-pick-of-#11428-upstream-release-5.14-1563826422' set up to track remote branch 'release-5.14' from 'upstream'.
+++ Downloading patch to /tmp/11428.patch (in case you need to do this again)

+++ About to attempt cherry pick of PR. To reattempt:
  $ git am -3 /tmp/11428.patch

Applying: MM-14139: Creating permissions for invite/promote/demote guests (#10778)
Applying: Adding invite guest api endpoint (#10792)
Using index info to reconstruct a base tree...
M	api4/team.go
M	api4/team_test.go
M	app/email.go
M	app/team.go
M	app/team_test.go
M	app/user.go
M	app/user_test.go
M	i18n/en.json
M	model/client4.go
M	store/sqlstore/tokens_store.go
M	store/sqlstore/upgrade.go
Falling back to patching base and 3-way merge...
Auto-merging store/sqlstore/upgrade.go
CONFLICT (content): Merge conflict in store/sqlstore/upgrade.go
Auto-merging store/sqlstore/tokens_store.go
Auto-merging model/client4.go
Auto-merging i18n/en.json
Auto-merging app/user_test.go
CONFLICT (content): Merge conflict in app/user_test.go
Auto-merging app/user.go
CONFLICT (content): Merge conflict in app/user.go
Auto-merging app/team_test.go
CONFLICT (content): Merge conflict in app/team_test.go
Auto-merging app/team.go
CONFLICT (content): Merge conflict in app/team.go
Auto-merging app/email.go
CONFLICT (content): Merge conflict in app/email.go
Auto-merging api4/team_test.go
Auto-merging api4/team.go
Patch failed at 0002 Adding invite guest api endpoint (#10792)
Use 'git am --show-current-patch' to see the failed patch
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".

+++ Conflicts detected:

UU app/email.go
UU app/team.go
UU app/team_test.go
UU app/user.go
UU app/user_test.go
UU store/sqlstore/upgrade.go

+++ Aborting in-progress git am.

+++ Returning you to the master branch and cleaning up.

```

mattermod commented:
Error doing the upgrade process. Please remove the label and try again.

jespino commented:
@ogi-m @srkgupta I copied all your bugs here to jira, please, any new bug create it on jira to track them better

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Spintmint test server creation failed. See the logs for more information.
